internal/ingress/metric/collectors/socket.go

```diff
@@ -221,8 +221,8 @@ func (sc *SocketCollector) handleMessage(msg []byte) {
 	}
 
 	for _, stats := range statsBatch {
-		if !sc.hosts.Has(stats.Host) {
-			klog.V(3).Infof("skiping metric for host %v that is not being served", stats.Host)
+		if sc.metricsPerHost && !sc.hosts.Has(stats.Host) {
```

Comment:
@choffmeister What do you think about adding a separate flag for this?
There are use cases when you need to get metrics for wildcard ingress and tag them with host.

Reply:
What kind of flag? As I understand my PR I did not change functionality but only fixed a bug. There are currently (and still are after this PR) two cases:

1. `--metrics-per-host=true` (default): In this case for wildcards there are no metrics recorded at all (to prevent DDOS)
2. `--metrics-per-host=false`: In this case there is no need ignore metrics for wildcard domains. But still (and this is what this PR fixes) there are currently no metrics collected for wildcards.

Reply:
Nothing is more permanent than a temporary solution ;)

There are 2 different actions that you're trying to control in this patch:
- collecting metrics for wildcard ingress
- labeling metric by host

In my opinion, those 2 are complementary:
- you may want not to collect metrics for the wildcard ingress, but label them by host
- you may need to collect metrics for the wildcard ingress, labeling them by host (e.g. in case hosts are filtered by an ALB or another LB, so you're preventing DOS at another level)
- and finally you may have your use case - collecting metrics for wildcard ingress, without labeling them by host
- etc

Thanks to your patch, I can use a fork and get metrics while changes will be merged into upstream. However, I think that final solution has to be a little more flexible.

Reply:
I disagree here: In my opinion the DDOS mitigation broke metrics for wildcards. With the already present `—metrics-per-host` (I did not add this flag, it already exists) there is no reason to not log metrics for wildcards. So in my eyes I fixed a regression that had been introduced with 0.20.0 (see the two issues that I linked). So I don’t see this as temporary solution but just as a bugfix.

But no problem if you want to solve it differently. I am fine with running my custom docker image until then.

Reply:
I disagree here: In my opinion the DDOS mitigation broke metrics for wildcards. With the already present `—metrics-per-host` (I did not add this flag, it already exists) there is no reason to not log metrics for wildcards. So in my eyes I fixed a regression that had been introduced with 0.20.0 (see the two issues that I linked). So I don’t see this as temporary solution but just as a bugfix.

But no problem if you want to solve it differently. I am fine with running my custom docker image until then.

Reply:
As our monitoring/alerting setup stopped working with versions >= 0.20 due to the wildcard metric filtering it would mean a bugfix for us.